### PR TITLE
Feature/python poetry support

### PR DESCRIPTION
**Pull request by meghfossa**
# Overview

Adds Poetry Support for Python projects.

I've closed previous PR due to significant noise - this addresses all the comments and nits. It should be easier to browse as well. Closed PR link is: https://github.com/fossas/spectrometer/pull/284
 
## Acceptance criteria

_What is this PR trying to accomplish?_

* User can get direct dependencies with `pyproject.toml`
* User can get direct and dependencies with `pyproject.toml` and `poetry.lock`
* User can use different package sources (git, url, etc.) 

## Testing plan

_How did you validate that this PR works? How did you check that the acceptance criteria were fulfilled?_

* I wrote unit tests, as well as manually tested against example projects with `fossa analyze --output` and `poetry show --tree --no-dev`
* I have performed end-to-end test with core

_This section should list reproduction steps that a reviewer can run through (and provide any needed test cases)._

* You can clone `https://github.com/meghfossa/example-projects` repository, it has example python poetry projects

## Risks

_Highlight any areas that you're unsure of, or want reviewers to pay particular attention to._

* I think some of my code is fairly verbose, I'm looking to get some feedback on my approach, before I can refractor/refine 
* I'm not sure, how best to handle case insensitive nature of poetry pkg, I have went with primitive approach for now

## References

_List any referenced GitHub issues. If PR references tickets in other systems (e.g. Zendesk), they should probably be mirrored as a GitHub Issue._

This closes fossas/team-analysis#549

## Checklist

- [x] I added tests for this PR's change (or confirmed tests are not viable).
- [x] I linted and formatted (via `haskell-language-server`) any haskell files I touched in this PR.
- [x] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [ ] I updated `Changelog.md` if this change is externally facing. If this PR did not mark a release, I added my changes into an `# Unreleased` section at the top.
- [x] I linked this PR to any referenced GitHub issues, if they exist.

### CODE REVIEW

File: test/Python/Poetry/PyProjectSpec.hs
Comment (meghfossa) [inline test/Python/Poetry/PyProjectSpec.hs:66]: Note this gets translated into `1.1` when heading to the backend. Our backend unable to solve version constraint is added as limitation in poetry doc (fyi: we are unable resolve version only when poetry.lock does not exist)

**Comment by scruffystuffs** [comment]
I'm not sure why CI isn't running, but I am curious why you forked and submitted a remote PR, instead of a local merge PR.  I think that has something to do with it.

**Comment by meghfossa** [comment]
> I'm not sure why CI isn't running, but I am curious why you forked and submitted a remote PR, instead of a local merge PR. I think that has something to do with it.

I prefer to have forked repository for work instead of directly working against a source. It makes git flow simpler. Interesting that CI is not picking this up. I did allow write permission to  maintainers. I can target this direct from upstream, but wait for review to finish. 

https://docs.github.com/en/actions/managing-workflow-runs/approving-workflow-runs-from-public-forks

**Comment by meghfossa** [comment]
It seems CI checks do not work with forks. Will have to redo this from the upstream origin instead. :disappointed: 

Will close this, and recreate PR again, if the CI checks do not start.

**Review by scruffystuffs — COMMENTED** [review]
Test

**Review by scruffystuffs — DISMISSED** [review]
test

**Comment by scruffystuffs** [comment]
Yeah, CI isn't working for your fork.

**Comment by meghfossa** [comment]
Side tracking, but we may need: `pull_request` for GH workflow, instead of push only  https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-events-for-forked-repositories

**Comment by scruffystuffs** [comment]
Yep, now the github token is messing with us.  During the build, we pull code from private repos in the org to embed executables.  We can give forks full access to THIS repo, but that doesn't help get them access to the OTHER repos while running from a fork.  For now, forking is not viable, sorry.

**Comment by scruffystuffs** [comment]
See here for how we embed the files: https://github.com/fossas/spectrometer/blob/5a06701440cc2062f3359ef9373fb28c2d03bee6/src/App/Fossa/EmbeddedBinary.hs#L125-L136

And here for the downloading process: https://github.com/fossas/spectrometer/blob/5a06701440cc2062f3359ef9373fb28c2d03bee6/vendor_download.sh

**Pull request closed without merging**
